OpenStreetMapViewer/src/main/java/org/osmdroid/samplefragments/drawing/ShowAdvancedPolylineStylesInvalidation.java

```diff
@@ -0,0 +1,208 @@
+        points.add(new GeoPoint(39.5, -11.5));
+
+        // create new polyline
+        mPolyline = new Polyline(mMapView, false, false);
+
+        // setup a black border
+        final Paint paintBorder = new Paint();
+        paintBorder.setColor(Color.BLACK);
+        paintBorder.setAntiAlias(true);
+        paintBorder.setStrokeWidth(25);
+        paintBorder.setStyle(Paint.Style.STROKE);
+        paintBorder.setStrokeJoin(Paint.Join.ROUND);
+        paintBorder.setStrokeCap(Paint.Cap.ROUND);
+        paintBorder.setAntiAlias(true);
+        mPolyline.getOutlinePaintLists().add(new MonochromaticPaintList(paintBorder));
+
+        // add points to line
+        mPolyline.setPoints(points);
+
+        // fill list with scalars
+        scalars.clear();
+        scalars.add(0.0f);
+        scalars.add(20.0f);
+        scalars.add(10.0f);
+        scalars.add(30.0f);
+        scalars.add(50.0f);
+        scalars.add(25.0f);
+
+        // setup mapping objects
+        mMapping = new ColorMappingVariationHue(0,50, 0,120, 1.0f, 0.5f);
+        mContainer = new ColorMappingForScalarContainer(mMapping);
+
+        // add scalars to mapping and container
+        // you need to add the scalars to both lists
+        for (final float scalar : scalars) {
+            mMapping.add(scalar);
+            mContainer.add(scalar);
+        }
+
+        // setup the mapping
+        final Paint paintMapping = new Paint();
+        paintMapping.setAntiAlias(true);
+        paintMapping.setStrokeWidth(20);
+        paintMapping.setStyle(Paint.Style.FILL_AND_STROKE);
+        paintMapping.setStrokeJoin(Paint.Join.ROUND);
+        paintMapping.setStrokeCap(Paint.Cap.ROUND);
+        paintMapping.setAntiAlias(true);
+        mPolyline.getOutlinePaintLists().add(new PolychromaticPaintList(paintMapping, mMapping, true));
+
+        // update UI
+        mMapView.getOverlayManager().add(mPolyline);
+        // force a redraw (normally triggered when map is moved for example)
+        mMapView.invalidate();
+        textViewCurrentLocation.setText("Scalar range from 0 to 50\nfor hue ranging from 0 to 120.");
+        btnProceed.setText("Extend Polyline");
+    }
+
+    /*
+     * Extends the line and invalidates the mapping with a new range.
+     */
+    private void extendAndInvalidateLine() {
+
+        // add new points
+        mPolyline.addPoint(new GeoPoint(40.0, -11.0));
+        mPolyline.addPoint(new GeoPoint(40.5, -11.5));
+        mPolyline.addPoint(new GeoPoint(41.0, -11.0));
+        mPolyline.addPoint(new GeoPoint(41.5, -11.5));
+        // add scalars with higher values to mapping and container
+        mMapping.add(80.f);
+        mMapping.add(60.f);
+        mMapping.add(100.f);
+        mMapping.add(100.f);
+        mContainer.add(80.f);
+        mContainer.add(60.f);
+        mContainer.add(100.f);
+        mContainer.add(100.f);
+
+        // update mapping with scalar end updated from 50 to 100
+        mMapping.init(0,100, 0,120);
```

Comment:
Please don't use magic numbers: why 120, why 100?
* If 120 means something, create a dedicated constant like, `static final int MAX_HUE = 120;` (or double, I don't know)
* Same for 100: I should see `MAX_SCALAR`, or even a value computed from the scalar list (as the max value)

Reply:
I wanted to display "speed" ranging from red to green. That is 0 to 120:: https://i.stack.imgur.com/YOBFy.png